### PR TITLE
Fix description of TypedArray.from()

### LINKED ISSUE #26729: Description of the differences between TypedArray.from() and Array.from() are not correct.
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/from

### What specific section or headline is this issue about?

Differences from Array.from()

### What information was incorrect, unhelpful, or incomplete?

- > If the thisArg value passed to TypedArray.from() is not a constructor, TypedArray.from() will throw a [TypeError](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypeError), where Array.from() defaults to creating a new [Array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array).

It turns out that the `thisArg` in the arguments list of `TypedArray.from(arrayLike, mapFn, thisArg)` can be a non-constructor. For example: a plain object in the following code:

`const arr = Uint32Array.from([1, 2, 3], x => x, {});`

If you run this code, there won't be errors be thrown.

I think the author meant not the `thisArg` passed to `TypedArray.from()` but the thing based on which `TypedArray.from()` is called. In other words, the receiver of `TypedArray.from()`.

So if we change the receiver manually like this:

`Uint32Array.from.call({}, [1, 2, 3]);`

We can find the following type error.

image

- > When the source parameter is an iterator, the TypedArray.from() first collects all the values from the iterator, then creates an instance of thisArg using the count, then sets the values on the instance. Array.from() sets each value as it receives them from the iterator, then sets its length at the end. 

I think the `source` should be `arrayLike`, right?
 

### What did you expect to see?

- If the `this` value based on which TypedArray.from() is called is not a constructor, TypedArray.from() will throw a [TypeError](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypeError), where Array.from() defaults to creating a new [Array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array).

- When the `arrayLike` parameter is an iterator, the TypedArray.from() first collects all the values from the iterator, then creates an instance of thisArg using the count, then sets the values on the instance. Array.from() sets each value as it receives them from the iterator, then sets its length at the end. 

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/javascript/reference/global_objects/typedarray/from`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/from
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/javascript/reference/global_objects/typedarray/from/index.md
* Last commit: https://github.com/mdn/content/commit/c51736321bee582320a6aed38f969e9f28a2fcd8
* Document last modified: 2023-03-30T14:41:11.000Z

</details>

### PR DESCRIPTION
**Pull request by Josh-Cena**
### Description

Fix https://github.com/mdn/content/issues/26729

### CODE REVIEW

File: files/en-us/web/javascript/reference/global_objects/typedarray/from/index.md [hunk 1]
```diff
@@ -54,43 +54,13 @@ A new {{jsxref("TypedArray")}} instance.
...
-  value as it receives them from the iterator, then sets its `length` at
-  the end.
-- When `Array.from()` gets an array-like which isn't an iterator, it
-  respects holes. `TypedArray.from()` will ensure the result
-  is dense.
+See {{jsxref("Array.from()")}} for more details. There are some subtle distinctions between {{jsxref("Array.from()")}} and `TypedArray.from()` (note: the `this` value mentioned below is the `this` value that `TypedArray.from()` was called with, not the `thisArg` argument used to invoke `mapFn`):
+
+- If the `this` value of `TypedArray.from()` is not a constructor, `TypedArray.from()` will throw a {{jsxref("TypeError")}}, while `Array.from()` defaults to creating a new {{jsxref("Array")}}.
+- The object constructed by `this` must be a `TypedArray` instance, while `Array.from()` allows its `this` value to construct any object.
+- When the `source` parameter is an iterator, the `TypedArray.from()` first collects all the values from the iterator, then creates an instance of `this` using the count, then sets the values on the instance. `Array.from()` sets each value as it receives them from the iterator, then sets its `length` at the end.
```
Comment (Elchi3) [inline files/en-us/web/javascript/reference/global_objects/typedarray/from/index.md:61]:
```suggestion
- When the `source` parameter is an iterator, `TypedArray.from()` first collects all the values from the iterator, then creates an instance of `this` using the count, and finally sets the values on the instance. `Array.from()` sets each value as it receives them from the iterator, then sets its `length` at the end.
```

File: files/en-us/web/javascript/reference/global_objects/typedarray/from/index.md [hunk 2]
```diff
@@ -54,43 +54,13 @@ A new {{jsxref("TypedArray")}} instance.
...
-  count, then sets the values on the instance. `Array.from()` sets each
-  value as it receives them from the iterator, then sets its `length` at
-  the end.
-- When `Array.from()` gets an array-like which isn't an iterator, it
-  respects holes. `TypedArray.from()` will ensure the result
-  is dense.
+See {{jsxref("Array.from()")}} for more details. There are some subtle distinctions between {{jsxref("Array.from()")}} and `TypedArray.from()` (note: the `this` value mentioned below is the `this` value that `TypedArray.from()` was called with, not the `thisArg` argument used to invoke `mapFn`):
+
+- If the `this` value of `TypedArray.from()` is not a constructor, `TypedArray.from()` will throw a {{jsxref("TypeError")}}, while `Array.from()` defaults to creating a new {{jsxref("Array")}}.
+- The object constructed by `this` must be a `TypedArray` instance, while `Array.from()` allows its `this` value to construct any object.
```
Comment (Elchi3) [inline files/en-us/web/javascript/reference/global_objects/typedarray/from/index.md:60]: Maybe?

```suggestion
- The object constructed by `this` must be a `TypedArray` instance, while `Array.from()` allows its `this` value to be constructed to any object.
```

File: files/en-us/web/javascript/reference/global_objects/typedarray/from/index.md [hunk 3]
```diff
@@ -54,43 +54,13 @@ A new {{jsxref("TypedArray")}} instance.
...
-- When the `source` parameter is an iterator, the
-  `TypedArray.from()` first collects all the values from the
-  iterator, then creates an instance of `thisArg` using the
-  count, then sets the values on the instance. `Array.from()` sets each
-  value as it receives them from the iterator, then sets its `length` at
-  the end.
-- When `Array.from()` gets an array-like which isn't an iterator, it
-  respects holes. `TypedArray.from()` will ensure the result
-  is dense.
+See {{jsxref("Array.from()")}} for more details. There are some subtle distinctions between {{jsxref("Array.from()")}} and `TypedArray.from()` (note: the `this` value mentioned below is the `this` value that `TypedArray.from()` was called with, not the `thisArg` argument used to invoke `mapFn`):
```
Comment (Elchi3) [inline files/en-us/web/javascript/reference/global_objects/typedarray/from/index.md:57]: Maybe give the fact that this is similar to Array.from() some more room to breathe?

```suggestion
The {{jsxref("Array.from()")}} method works similar; see its documentation for more details. 

There are some subtle distinctions between {{jsxref("Array.from()")}} and `TypedArray.from()` (note: the `this` value mentioned below is the `this` value that `TypedArray.from()` was called with, not the `thisArg` argument used to invoke `mapFn`):
```

**Review by Elchi3 — APPROVED** [review]
Thanks, some optional nits. LGTM

File: files/en-us/web/javascript/reference/global_objects/typedarray/from/index.md (hunk 3 above)
Comment (Josh-Cena) [reply files/en-us/web/javascript/reference/global_objects/typedarray/from/index.md:57]: I plan to add the same structure to all other TypedArray pages ("see [the array method] for more details"), so what about just adding a line break?

```suggestion
See {{jsxref("Array.from()")}} for more details.

There are some subtle distinctions between {{jsxref("Array.from()")}} and `TypedArray.from()` (note: the `this` value mentioned below is the `this` value that `TypedArray.from()` was called with, not the `thisArg` argument used to invoke `mapFn`):
```

**Review by Elchi3 — APPROVED** [review]
Thanks @Josh-Cena 👍

**Pull request merged**